### PR TITLE
Added Support of Xlabs controlnet to FluxControlNetInpaintPipeline

### LINKED ISSUE #9733: Xlabs controlnets do not work with FluxControlNetInpaintPipeline
### Describe the bug

errors when Xlabs controlnets are used with FluxControlNetInpaintPipeline

### Reproduction

controlnet = FluxControlNet.from_pretrained(“xlabs…”)
pipe = FluxControlNetInpaintPipeline.from_pretrained(…, controlnet=controlnet)

image = pipe(…)

### Logs

```shell
Given groups=1, weight size of [16,3,3,3], expected input [1,1,4096,64] to have 3 channels, but got 1 channels instead.


Switching controlnet to a previously supported one such as InstantX does not result in this issue. All inputs are confirmed to be (h, w, 3) before forward pass.
```

### System Info

diffusers from source

### Who can help?

@sayakpaul @ang

### PR DESCRIPTION
**Pull request by SahilCarterr**
# What does this PR do?

Fixes #9733  Adds Support of Xlabs controlnet to FluxControlNetInpaintPipeline

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed.
@yiyixuxu 

-----
### Example
```python
import torch
from diffusers import FluxControlNetInpaintPipeline
from diffusers.models import FluxControlNetModel
from diffusers.utils import load_image

controlnet = FluxControlNetModel.from_pretrained(
  "Xlabs-AI/flux-controlnet-canny-diffusers",
  torch_dtype=torch.float16,
  use_safetensors=True,
)

pipe = FluxControlNetInpaintPipeline.from_pretrained(
    "black-forest-labs/FLUX.1-dev", controlnet=controlnet, torch_dtype=torch.float16
)

pipe.to("cuda")

prompt = "A woman wearing a red dress with the text 'Flux' on it and cat print"

image = pipe(
    prompt,
    image=init_image,
    mask_image=mask_image,
    control_image=canny_image,
    height=960,
    width=640,
    num_inference_steps=50,
    strength=0.8,
    guidance_scale=6.5,
    controlnet_conditioning_scale=0.7,
    control_guidance_start=0.2,
    control_guidance_end=0.8,
    generator=torch.Generator(device="cuda").manual_seed(42),
).images[0]
```
| Input Image    | Mask Image     | Canny Image    | Output Image    |
|----------------|----------------|----------------|-----------------|
| girl_8 | maskofgirl_8 | canny_example_girl8 | girl_08_output |

### CODE REVIEW

**Review by yiyixuxu — APPROVED** [review]
nice!
